operations.md added

mkurzman opened the pull request:
as discussed in https://github.com/spdx/spdx-3-model/pull/897

umm0 committed:
Add Operations profile draft files

Signed-off-by: Ummo Schwarting <69399030+umm0@users.noreply.github.com>

umm0 committed:
Add Operations profile draft files

Signed-off-by: Ummo Schwarting <69399030+umm0@users.noreply.github.com>

umm0 committed:
add some data types, rearrange markdown

umm0 committed:
add delivery class

mkurzman committed:
Merge branch 'spdx:profile-operations' into profile-operations

umm0 committed:
Merge pull request #890 from umm0/profile-operations

Profile operations, merge current discussion state as of 24-09-27
Merged after Operations profile meeting

umm0 committed:
Operations profile meeting 2024-10-01

mkurzman committed:
Merge branch 'spdx:profile-operations' into profile-operations

mkurzman committed:
Initial creation of properties and classes

mkurzman committed:
update the classes for review meeting

mkurzman committed:
intial empty files as base

mkurzman committed:
Merge branch 'spdx:main' into mkurzman_20241011

zvr committed:
Merge pull request #897 from boschglobal/mkurzman_20241011

mkurzman committed:
First draft of an Operations.md 

as base for discussion

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
updates in the descriptions

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

goneall committed:
Document branches

This is pretty much a copy / past from the SPDX Spec repo README Branch Structure section.

Signed-off-by: Gary O'Neall <gary@sourceauditor.com>

goneall committed:
Update per suggestion

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Gary O'Neall <gary@sourceauditor.com>

zvr committed:
Merge pull request #945 from spdx/goneall-patch-2

Document branches

bact committed:
Add notes on English as normative language

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

bact committed:
Merge pull request #946 from bact/add-english-normative-notes

Add notes on English as normative language

kestewart commented on model/Operations/Operations.md:
```diff
@@ -0,0 +1,22 @@
+
+## Summary
+
+The Operations Profile defines fields for describing the business context of the software that cannot (or not yet) directly extracted from the source repository.
```
```suggestion
The Operations Profile defines fields for describing the business context of the software that cannot (or not yet) be directly extracted from the source repository.
```

kestewart commented on model/Operations/Operations.md:
```diff
@@ -0,0 +1,22 @@
+
+The intention of the Operations Profile is to provide a common base rather for alignment of tool and infrastructure interfaces and development than for exchanging the data between organizations along the software supply chain. The managed information is expected to be mainly kept within the organizations but in some cases might also be necessary input for processing and determining relevant param ...
+
+The information about Applications shall describe facts about the distribution and business context of a product brought into the market, that could consist of several deliverables. There relation between Application and Deliverable can be 1:n.
```
```suggestion
The information about Element shall describe facts about the distribution and business context of a product brought into the market, that could consist of several deliverables. The relation between Element and Deliverable targets can be one to many.
```

kestewart replied:
Applications could be Products, Projects, Packages, etc.   so probably best to stick with Element term which could be any of these.

kestewart reviewed (commented):
Good starting point.   Couple of comments.

bact commented:
A bunch of type mismatch errors:

```text
ERROR:root:In class /Operations/DeliverableFacts, property programmingLanguage has type tbd but the range of /Operations/programmingLanguage is xsd:string
ERROR:root:In class /Operations/DeliverableFacts, property dependencyManager has type tbd but the range of /Operations/dependencyManager is xsd:string
ERROR:root:In class /Operations/DeliverableFacts, property packageManager has type tbd but the range of /Operations/packageManager is xsd:string
ERROR:root:In class /Operations/DeliverableFacts, property environmentFramework has type tbd but the range of /Operations/environmentFramework is xsd:string
ERROR:root:In class /Operations/DeliverableFacts, property applicationCategory has type tbd but the range of /Operations/applicationCategory is xsd:string
ERROR:root:In class /Operations/DeliverableFacts, property applicationType has type tbd but the range of /Operations/applicationType is xsd:string
ERROR:root:In class /Operations/DeliverableFacts, property distributionMethod has type tbd but the range of /Operations/distributionMethod is xsd:string
ERROR:root:In class /Operations/DeliverableFacts, property operatingSystem has type tbd but the range of /Operations/operatingSystem is xsd:string
ERROR:root:In class /Operations/DeliverableFacts, property consistsOf has type tbd but the range of /Operations/consistsOf is xsd:string
ERROR:root:In class /Operations/DeliverableFacts, property developedBy has type tbd but the range of /Operations/developedBy is xsd:string
ERROR:root:In class /Operations/DeliverableFacts, property contact has type tbd but the range of /Operations/contact is xsd:string
ERROR:root:In class /Operations/DeliverableFacts, property linkToArchitecture has type tbd but the range of /Operations/linkToArchitecture is xsd:string
ERROR:root:In class /Operations/DeliverableFacts, property osmConcept has type tbd but the range of /Operations/osmConcept is xsd:string
```

dependabot[bot] committed:
Bump actions/setup-python from 5.3.0 to 5.4.0

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 5.3.0 to 5.4.0.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/0b93645e9fea7318ecaed2b359559ac225c90a2b...42375524e23c412d93fb67b49958b491fce71c38)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mkurzman committed:
Update of the types

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Types in DeliverableFacts updated

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Transferred DeliverableFacts fields from Deliverables.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Transferred DeliverableFacts fields from Deliverables.md to delivery.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
fix - move supplierdeliverableFacts in supplierdeliverableFacts.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Moved supplierDeliverableFacts from deliverable Facts and updated description

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Bugfix - words mixed up

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Merge branch 'spdx:develop' into mkurzman_20241206

mkurzman committed:
Operations tab added by copy-paste

from source: https://github.com/umm0/spdx-3-model/blob/ummo-25-01-24/docs/model.drawio
line 1630 to 1939
insterted at the end before the closing tab

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

umm0 commented:
Hi,

any thought on merging this pull request in basically the sate it is currently in?

I would suggest doing this because currently we have the problem, that the 'profile-operations' branch is far behind our discussion state.
Because of company compliance reasons it is a bit complicated for @mkurzman and me to merge our draft states outside of this repo. Merging this pull request now could help us get back onto a common working base.

bact commented:
I think it should be fine to merge this to the `profile-operations` branch.

bact commented on model/Operations/Classes/ApplicationFacts.md:
````diff
@@ -0,0 +1,95 @@
+## Description
+
+The Application Facts are collected all along the product lifecyle and contents may be updated when the product reaches a new phase.
+```
````
```suggestion
```

This code block formatting should be removed.

bact commented on model/Operations/Properties/contact.md:
Potential overlap with `mailingAddress` proposed in Hardware profile
- https://github.com/spdx/spdx-3-model/pull/977

mkurzman replied:
Thank you for pointing this out. This was exactly the idea to bring this proposal into the model to cross-check with other profiles. But content-wise I would suggest to the HBOM to rethink the property to also be more generic, as Mail is not the only way of contacting and might be outdated some day.

bact commented on model/Operations/Vocabularies/applicationType.md:
```diff
@@ -0,0 +1,31 @@
+
+- name: applicationType
+
+## Entries
+
+- BT01 Embedded system and or application software in mass production: tbd
+- BT02 Embedded application software in mass production with FOTA: tbd
+- BT03 Embedded system software on smart devices with FOTA: tbd
+- BT04 Embedded application software on smart devices with FOTA: tbd
+- BT05 Client Application: tbd
+- BT06 Web Application: The Software is typically provided on a web-server and is interacting with the user via a browser. 
+- BT07 Server-based system software: tbd
+- BT08 Server-based application software: tbd
+- BT09 Cloud Service on customer premise: tbd
+- BT10 Cloud Service in the internet: tbd
+- BT11 Open Source Development Services: tbd
+- BT12 Source Code Sharing: tbd
```
Entry names should be in `lowerCamelCase`, without any space inside.

bact commented on model/Operations/Vocabularies/applicationType.md:
```diff
@@ -0,0 +1,31 @@
+SPDX-License-Identifier: Community-Spec-1.0
+
+# applicationType
```
```suggestion
# ApplicationType
```

Vocab name should be in `UpperCamelCase`.

From the summary and description, would it be more specific to use the name `DistributionType` or `DeploymentType` instead?

bact commented on model/Operations/Classes/DeliverableFacts.md:
```diff
@@ -0,0 +1,103 @@
+  - maxCount: 1
+
+- applicationType
+  - type: applicationTypeType
```
Has to be updated.

bact commented on model/Operations/Classes/DeliverableFacts.md:
```diff
@@ -0,0 +1,103 @@
+  - minCount: 1
+  - maxCount: n
+
+- contact
+  - type: Agent
+  - minCount: 1
+  - maxCount: n
+
+- linkToArchitecture
+  - type: xsd:anyURI
+  - minCount: 1
+  - maxCount: 1
+
+- osmConcept
+  - type: xsd:anyURI
+  - minCount: 1
+  - maxCount: 1
+
+- fossComplianceBundleStorage
+  - type: fossComplianceBundleStorageType
+  - minCount: 1
+  - maxCount: 1
+
+- reviews
+  - type: xsd:anyURI
+  - minCount: 1
+  - maxCount: n
+
+- comment
+  - type: xsd:string
+  - maxCount: 1
+
+- supplierDeliverableFacts
+  - type: Delivery
+  - minCount: 0
+  - maxCount: n
```
```suggestion
- contact
  - type: Agent
  - minCount: 1
- linkToArchitecture
  - type: xsd:anyURI
  - minCount: 1
  - maxCount: 1
- osmConcept
  - type: xsd:anyURI
  - minCount: 1
  - maxCount: 1
- fossComplianceBundleStorage
  - type: xsd:string
  - minCount: 1
  - maxCount: 1
- reviews
  - type: xsd:anyURI
  - minCount: 1
- comment
  - type: xsd:string
  - maxCount: 1
- supplierDeliverableFacts
  - type: Delivery
  - minCount: 0
```

There should be no newlines between property entries.

bact commented on model/Operations/Classes/DeliverableFacts.md:
```diff
@@ -0,0 +1,103 @@
+  - type: operatingSystemType
+  - minCount: 1
+  - maxCount: 1
+
+- consistsOf
+  - type: Artifact
+  - minCount: 1
+  - maxCount: n
```
```suggestion
- consistsOf
  - type: Artifact
  - minCount: 1
```

- There should be no newlines between property entries
- If an unbounded maximum number is desired, use `*` (see: https://spdx.github.io/spdx-spec/v3.0.1/conformance/ )
  - `maxCount: *` 
- maxCount is `*` by default, so that line can be also omitted

bact commented on model/Operations/Classes/DeliverableFacts.md:
```diff
@@ -0,0 +1,103 @@
+  - type: Artifact
+  - minCount: 1
+  - maxCount: n
+
+- developedBy
+  - type: Agent
+  - minCount: 1
+  - maxCount: n
```
```suggestion
- developedBy
  - type: Agent
  - minCount: 1
```

bact commented:
> any thought on merging this pull request in basically the sate it is currently in?
> 
> I would suggest doing this because currently we have the problem, that the 'profile-operations' branch is far behind our discussion state. Because of company compliance reasons it is a bit complicated for @mkurzman and me to merge our draft states outside of this repo. Merging this pull request now could help us get back onto a common working base.

@umm0 I think it should be fine to merge this to `profile-operations` branch. The branch is meant for development.

In order to do so, however, conflicts have to be resolved first.

Also, if possible, some obvious changes, like typo and formatting, can be made before the merge. Just not to lost it.
- Suggested changes can be rejected. Just give a comment and hit "Resolve conversation". So we can track that.

mkurzman committed:
Update model/Operations/Classes/ApplicationFacts.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update model/Operations/Classes/DeliverableFacts.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update model/Operations/Classes/DeliverableFacts.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update model/Operations/Classes/DeliverableFacts.md

Co-authored-by: Arthit Suriyawongkul <arthit@gmail.com>
Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update model/Operations/Operations.md

Co-authored-by: Kate Stewart <13152682+kestewart@users.noreply.github.com>
Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update model/Operations/Operations.md

Co-authored-by: Kate Stewart <13152682+kestewart@users.noreply.github.com>
Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update applicationType.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update and rename applicationType.md to ApplicationType.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
UpperCase for Vocabularies

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman commented:
Hi, Thank you for all the hints. I tried to integrated all valuable suggested changes.

mkurzman commented:
It seems I do not have the necessary rights to resolve the conflicts.

mkurzman committed:
empty lines removed

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update ApplicationFacts.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
remove empty lines

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
supplyChain Context as optional

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

bact reviewed (approved):
Good for now. Suggest merge as soon as possible once conflicts are resolved, so the Profile team can have a common branch to work upon.

mkurzman committed:
Merge branch 'profile-operations' into mkurzman_20241206

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

bact commented:
No conflicts now 🎉 

However, there still 5 errors from PR validation, most of them are about type mismatches:

```
Nested list parsing error in line `tbd'
In class /Operations/ApplicationFacts, property productOwner has type /Core/Person but the range of /Operations/productOwner is xsd:string
In class /Operations/ApplicationFacts, property documentationLink has type xsd:anyURI but the range of /Operations/documentationLink is xsd:string
In class /Operations/ApplicationFacts, property productAccessURL has type xsd:anyURI but the range of /Operations/productAccessURL is xsd:string
In class /Operations/ApplicationFacts, property distributedDeliverables has type Artifact but the range of /Operations/distributedDeliverables is xsd:string
```

mkurzman commented:
Where does the validator take the "range" reference from to detect this type mismatch? So where to fix the wrong "range"?

mkurzman committed:
Update productOwner.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update documentationLink.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update productAccessURL.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update distributedDeliverables.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update ApplicationFacts.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update DeliverableFacts.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update Delivery.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update SupplierDeliverableFacts.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update distributedDeliverables.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update ApplicationFacts.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update technicalDeployment.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update ApplicationFacts.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update contact.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update scope.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update relationType.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update supplyChainContext.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update releaseCycles.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update fossComplianceBundleProvision.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update contractSetup.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update ApplicationFacts.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update fossTermsTowardsCustomer.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update distributionTermsTowardsCustomer.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update ApplicationFacts.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update programmingLanguage.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update dependencyManager.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update packageManager.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update environmentFramework.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update applicationCategory.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update applicationType.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update distributionMethod.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update operatingSystem.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update DeliverableFacts.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update consistsOf.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update developedBy.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update linkToArchitecture.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update osmConcept.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Create fossComplianceBundleStorage.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Create reviews.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update DeliverableFacts.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update SupplierDeliverableFacts.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update supplierName.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update deliverableFromSupplier.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update fossTermsTowardsSupplier.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update distributionTermsFromSupplier.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update fossComplianceBundleConsumption.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman committed:
Update supplierFossContact.md

Signed-off-by: Marcel Kurzmann <marcel.kurzmann@bosch.com>

mkurzman commented:
May we perhaps deactivate this PR validator for this PR as we need this Dev-Branch to collaborate, thus we need to incrementally improve it. If I have to get every little change through the validator it is a little bit cumbersome as potentially we will delete or replace properties and vocabularies anyway. The validator is helpful once we have a more stable phase, but we still need an incubation mode that enables us collaboration in this profile-operations-dev-branch.

bact commented:
I think for this initial stage it's fine to merge with the validation errors.

Let's bring this up during the Tech call today and see how people think.

zvr commented:
The "validator" only checks obvious errors, like mismatched types or (the current error) that a property "reviews" does not say its name is "relationType". I think such errors should be fixed as soon as possible, since otherwise the files do not make sense.

If the issue is that the validator runs after a commit is pushed and reports a single error and stops, I agree this is inconvenient. To fix this:
- the validator should be run manually before commit/push, to check the input
- I'll push the changes next week that it will report _all_ errors in one run, rather than stopping at each one. (there's a major overhaul of all reporting)

bact commented:
4 March 2025 Tech call agrees to have this PR merge to the `spdx:profile-operations` branch and any needed fix can be followed within that branch.

bact commented on model/Operations/Properties/reviews.md:
```diff
@@ -0,0 +1,17 @@
+
+## Metadata
+
+- name: relationType
```
```suggestion
- name: reviews
```

This much matched with the property name "reviews" (the name of the top heading and the filename).

(Note that we may need to call it all a singular "review", instead of a plural "reviews", because of a decision in https://github.com/spdx/spdx-3-model/issues/226

bact replied:
@mkurzman I agree with zvr that at the very least this obvious error of name mismatch should get fixed here in this PR.

(the change to singular form can be done later, however)

bact commented on model/Operations/Vocabularies/ApplicationType.md:
```diff
@@ -0,0 +1,31 @@
+
+## Entries
+
+- embeddedSystemAndOrApplicationsSoftwareInMassProduction: BT01 Embedded system and or application software in mass production
+- embeddedApplicationSoftwareInMassProductionWithFOTA: BT02 Embedded application software in mass production with FOTA
+- embeddedSystemSoftwareOnSmartDevicesWithFOTA: BT03 Embedded system software on smart devices with FOTA
+- embeddedApplicationSoftwareOnSmartDevicesWithFOTA: BT04 Embedded application software on smart devices with FOTA
+- clientApplication: BT05 Client Application
+- webApplication: BT06 Web Application - The Software is typically provided on a web-server and is interacting with the user via a browser. 
+- serverBasedSystemSoftware: BT07 Server-based system software
+- serverBasedApplicationSoftware: BT08 Server-based application software
+- cloudServiceOnPremise: BT09 Cloud Service on premise in private cloud
+- cloudServiceInTheInternet: BT10 Cloud Service in the internet
+- openSourceDevelopmentService: BT11 Open Source Development Services
+- sourceCodeSharing: BT12 Source Code Sharing
```
```suggestion
- embeddedSystemAndOrApplicationsSoftwareInMassProduction: BT01 Embedded system and/or application software in mass production.
- embeddedApplicationSoftwareInMassProductionWithFOTA: BT02 Embedded application software in mass production with Firmware Over-the-Air (FOTA).
- embeddedSystemSoftwareOnSmartDevicesWithFOTA: BT03 Embedded system software on smart devices with  Firmware Over-the-Air (FOTA).
- embeddedApplicationSoftwareOnSmartDevicesWithFOTA: BT04 Embedded application software on smart devices with  Firmware Over-the-Air (FOTA).
- clientApplication: BT05 Client application.
- webApplication: BT06 Web application - the software is typically provided on a web-server and is interacting with the user via a browser. 
- serverBasedSystemSoftware: BT07 Server-based system software.
- serverBasedApplicationSoftware: BT08 Server-based application software.
- cloudServiceOnPremise: BT09 Cloud service on premise in private cloud.
- cloudServiceInTheInternet: BT10 Cloud service in the Internet.
- openSourceDevelopmentService: BT11 Open source development services.
- sourceCodeSharing: BT12 Source code sharing.
```

- Expands "FOTA" (Firmware Over-the-Air)
- Consistent fullstops
- Normal casing for non-pronouns or special terms

umm0 commented:
We agreed in the Operations Profile Meeting today (2025-03-07) to merge this pull request as is in order to create a common working base.
